### ISSUE TITLE
SceneNN Data link is broken

### ISSUE THREAD

**Issue by jean-noelp** [scenenn/pointwise#11]
Hello,

All in title : the link to download SceneNN Segmentation data is broken.
How to get this dataset so ?

thanks !

**Comment by songuke** [comment]
Thank you for reporting the issue.

I have updated the links. You can download at:
https://drive.google.com/file/d/1DcKKrdzirUjKejjasexGdU0ECMpIXCD3/view

or

http://103.24.77.34:8080/scenenn/home/cvpr18/data/scenenn_seg_76.zip


2019年1月24日(木) 1:51 jean-noelp <notifications@github.com>:

**Comment by swiss-knight** [comment]
Why not also updating it in the README.me ?

**Comment by sidml** [comment]
@songuke This gdrive link is currently working and has 76 scenes. **But the hdf5 files only have _coords_.** Please let me know from where can I obtain the semantic and instance labels. Thanks.

**Comment by sonhua** [comment]
Hi Siddhartha, I also uploaded the 76 scenes PLY files. Please see if this is what you need: https://drive.google.com/file/d/1d_ILfaxpJBpiiwCZtvC4jEKnixEr9N2l/view?usp=sharing

**Comment by sidml** [comment]
@sonhua 
Thanks for the quick reply. I downloaded the data from the shared link. It seems to contain ply files.
I observed that the number of points in ply file is different from h5 file. For example: 005.ply contains 1758578 points while the scenenn_coords_005.hdf5 file contains 7999488 points (651 x 4096 x 3). I was wondering if this is expected ?

If possible, it will be great if you can refer or share code to convert plyfile into h5 file, that is compatible with this repository. Thanks for your help.

**Comment by songuke** [comment]
Yes this is expected. The PLYs are the original data that I apply sampling
to create the training data for segmentation in the paper.

This is the code I used. FYI.

Note that it is not runnable as there is an internal library that I do not
release (due to the fact that I did not manage to get all permissions
needed.) But those are just geometry processing so it should not affect
reproducibility.

2020年10月20日(火) 14:11 Siddhartha <notifications@github.com>:

**Comment by sidml** [comment]
@songuke 
> This is the code I used. FYI.

Sorry, can you please share the code again. Somehow, it didn't get shared. Thanks.

**Comment by sonhua** [comment]
Sorry. I did not notice that sharing .py does not work. I changed to .txt here. 
[prepare_scenenn_seg.txt](https://github.com/hkust-vgd/pointwise/files/5415124/prepare_scenenn_seg.txt)

**Comment by sidml** [comment]
@sonhua 
Thanks for sharing the code. The code seems to be using internal library **libpluie**.  (I understand that it not possible to share this library. )
But, it seems to me that **libpluie** is required for generating the query box `query_box = pe.BoundingBox(np.append(query_box_center, box_size))` and subsequent downsampling `downsampled_mesh = submesh.grid_sample(n)`. So, right now it may not be possible to run the code on my end. 

I was wondering if it is possible for you to run the code and generate downsampled hdf5 files that contain semantic and instance labels.
